packages/table/src/table.tsx

```diff
@@ -477,29 +503,108 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
+                        className="bp-table-quadrant-scroll-container"
+                        ref={this.setMainQuadrantScrollRef}
+                        onScroll={this.handleMainQuadrantScroll}
+                    >
+                        <div className={Classes.TABLE_TOP_CONTAINER}>
+                            {this.renderMenu(this.setMainQuadrantMenuRef)}
+                            {this.renderColumnHeader()}
+                        </div>
+                        <div className={Classes.TABLE_BOTTOM_CONTAINER}>
+                            {this.renderRowHeader()}
+                            <div ref={this.setBodyRef} style={{ position: "relative" }}>
+                                {this.renderBody()}
+                            </div>
+                        </div>
+                    </div>
+                </div>
+                <div className="bp-table-quadrant-top" style={topQuadrantStyles} ref={this.setTopQuadrantRef}>
+                    <div
+                        className="bp-table-quadrant-scroll-container"
+                        ref={this.setTopQuadrantScrollRef}
+                        onWheel={this.handleTopQuadrantWheel}
+                    >
+                        <div className={Classes.TABLE_TOP_CONTAINER}>
+                            {this.renderMenu(this.setTopQuadrantMenuRef)}
+                            {this.renderColumnHeader()}
+                        </div>
+                        <div className={Classes.TABLE_BOTTOM_CONTAINER}>
+                            {this.renderRowHeader(0, numFrozenRows)}
+                            {this.renderBody(0, numFrozenRows, null, null, null, numFrozenRows)}
+                        </div>
+                    </div>
+                </div>
+                <div className="bp-table-quadrant-left" style={leftQuadrantStyles} ref={this.setLeftQuadrantRef}>
+                    <div
+                        className="bp-table-quadrant-scroll-container"
+                        ref={this.setLeftQuadrantScrollRef}
+                        onWheel={this.handleLeftQuadrantWheel}
+                    >
+                        <div className={Classes.TABLE_TOP_CONTAINER}>
+                            {this.renderMenu(this.setLeftQuadrantMenuRef)}
+                            {this.renderColumnHeader(0, numFrozenColumns)}
+                        </div>
+                        <div className={Classes.TABLE_BOTTOM_CONTAINER}>
+                            {this.renderRowHeader()}
+                            {this.renderBody(null, null, 0, numFrozenColumns, numFrozenColumns, null)}
+                        </div>
+                    </div>
                 </div>
-                <div className={Classes.TABLE_BOTTOM_CONTAINER}>
-                    {isRowHeaderShown ? this.renderRowHeader() : undefined}
-                    {this.renderBody()}
+                <div className="bp-table-quadrant-top-left" style={baseStyles} ref={this.setTopLeftQuadrantRef}>
```

Comment:
These are essentially four instances of the same pattern:

```tsx
<div className style ref>
    <div className="bp-table-quadrant-scroll-container" ref onWheel onScroll />
        <div className={Classes.TABLE_TOP_CONTAINER}>
            {this.renderMenu(ref)} // or maybe a menu placeholder for underlying quadrants? we just need one menu element to be visible...
            {this.renderColumnHeader(numFrozenColumns?)} // or maybe a placeholder for underlying quadrants?; can simplify these args
        </div>
        <div className={Classes.TABLE_BOTTOM_CONTAINER}>
            {this.renderRowHeader(numFrozenRows?)} // or a placeholder; can simplify these args
            {this.renderBody(numFrozenRows?, numFrozenColumns?)} // can simplify these args
        </div>
    </div>
</div>
```

Let's pull it out into a `TableQuadrant` component.